pybv/io.py

```diff
@@ -491,5 +492,11 @@ def _write_bveeg_file(eeg_fname, data, orientation, format, resolution, units):
         raise ValueError(msg)
     data = data.astype(dtype=dtype)
 
+    # Swap bytes if system architecure is big-endian and dtype is "native" or
+    # "big-endian"
+    if (sys.byteorder == "big" and
+       (data.dtype.byteorder == "=" or data.dtype.byteorder == ">")):
+        data = data.byteswap()
```

Comment:
I don't really get this if clause.

Shouldn't we swap bytes if `sys.byteorder == "big"` AND `data.dtype.byteorder == "="` ... OR `data.dtype.byteorder == ">"`

In your case, couldn't it be possible that we are on a little-endian machine the byte order is big-endian? --> then big-endian data would be written, because the if-clause does not get triggered?

@Aniket-Pradhan

Reply:
Theoretically this could happen, so this should probably be changed to:

```
if (sys.byteorder == "big" and data.dtype.byteorder == "=") or data.dtype.byteorder == ">":
        data = data.byteswap()
```

Reply:
OTOH, when will `data.dtype.byteorder == ">"` ever be `True` on a little Endian machine? The default is always `=` AFAIK, so checking for `sys.byteorder` might be completely sufficient, i.e.

```
if sys.byteorder == "big":
    data = data.byteswap()
```

Reply:
It's not usually ideal that the byteorder would be big-endian on a little-endian machine, but yes, we can swap the bytes for this scenario. I'll update this in a new commit.

Reply:
> OTOH, when will `data.dtype.byteorder == ">"` ever be `True` on a little Endian machine? The default is always `=` AFAIK, so checking for `sys.byteorder` might be completely sufficient, i.e.

This is true, but one cannot be too sure of what the user is passing `data` as. That's why I added in the checks. It may be redundant, but we can be confident that we have covered all the possible cases.